api/cmd/api/setup.go

```diff
@@ -106,54 +105,65 @@ func initImageBuilder(cfg *config.Config) (webserviceBuilder imagebuilder.ImageB
 
 	var artifactService artifact.Service
 	if cfg.ImageBuilderConfig.ArtifactServiceType == "gcs" {
-		gcsClient, err := gcs.NewClient(context.Background())
-		if err != nil {
-			log.Panicf("%s,failed initializing gcs for mlflow delete package", err.Error())
-		}
-		artifactService = artifact.NewGcsArtifactClient(gcsClient)
+		artifactService, err = artifact.NewGcsArtifactClient()
```

Comment:
The need to manually initialise a GCS client is gone with the refactoring done in https://github.com/caraml-dev/mlp/pull/116 (the imported `NewGcsArtifactClient` will do it instead).